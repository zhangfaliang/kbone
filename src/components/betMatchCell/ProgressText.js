import React from 'react';
import classnames from 'classnames';
import styles from './index.less';
const ProgressText = ({
  handleClick = () => {},
  stringInTime = '',
  text = '',
  disabled = false,
}) => {
  const handleClickProgress = event => {
    event.preventDefault();
    !disabled && handleClick && handleClick();
  };
  const stringInTimeCls = classnames({
    [styles.stringInTime]: text,
    [styles.stringTime]: !text,
  });

  return (
    <div className={styles.progress} onClick={handleClickProgress}>
      <span className={styles.text}>{text}</span>
      <span className={stringInTimeCls}>{stringInTime}</span>
    </div>
  );
};

ProgressText.COMPONENT_NAME = 'PROGRESSTEXT';
export default ProgressText;
