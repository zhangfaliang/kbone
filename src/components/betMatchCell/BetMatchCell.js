import React, { Component } from 'react';
import styles from './index.less';
import BetMatchInfo from '../betMatchInfo';
import BetButtonCell from '../betButtonCell';
import ProgressText from './ProgressText';
class BetMatchCell extends Component {
  static COMPONENT_NAME = 'BETMATCHCELL';
  handleBetBtn = betInfo => {
    const { matchTime, clickBetBtn } = this.props;
    clickBetBtn({ ...betInfo, matchTime });
  };
  render() {
    const { children, prefixCls } = this.props;
    const classNameStr = `${prefixCls || 'push-inplay'}-bet-cell`;
    const childrens = React.Children.map(children, option => {
      const { clickBetBtn, changeSp, changeHandicap, ...other } = option.props;

      if (option.type.COMPONENT_NAME === 'BETMATCHINFO') {
        return <BetMatchInfo {...other} />;
      } else if (option.type.COMPONENT_NAME === 'BETBUTTONCELL') {
        return (
          <BetButtonCell
            {...other}
            clickBetBtn={this.handleBetBtn}
            changeSp={this.props.changeSp}
            changeHandicap={this.props.changeHandicap}
          />
        );
      } else if (option.type.COMPONENT_NAME === 'PROGRESSTEXT') {
        //handleClick={this.props.betNow}
        return <ProgressText {...other} />;
      }
    });
    return <div className={styles[classNameStr]}>{childrens}</div>;
  }
}
BetMatchCell.defaultProps = {
  matchTime: '',
  betNow: () => {
    console.log('betNow');
  },
  clickBetBtn: params => {
    console.log(params);
  },
  changeSp: () => {
    console.log('changeSp');
  },
  changeHandicap: () => {
    console.log('changeHandicap');
  },
};

export default BetMatchCell;
