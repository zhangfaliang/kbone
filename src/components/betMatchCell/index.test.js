import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';

import BetMatchInfo from '../betMatchInfo';
import BetButtonCell from '../betButtonCell';
import ProgressText from './ProgressText';
import BetMatchCell from './BetMatchCell';
import BetButton from '../betButton';
import Line from '../line';
describe('betMatchCell', async () => {
  it(' toJson ProgressText', () => {
    const wrapper = render(
      <BetMatchCell
        matchTime={'werwer'}
        prefixCls={'push-inplay'}
        clickBetBtn={() => {
          console.log('test12');
        }}
      >
        <BetMatchInfo
          homeName={'678'}
          homeScore={34}
          awayName={'34'}
          awayScore={34}
          isLive={true}
          isShowScore={true}
          liveText="LIVE"
          time={'werwerwer'}
          clickBetMatchInfo={() => {
            this.goToMatchDetail({ sportType, matchId });
          }}
        />

        <ProgressText text={'ewrwe'} stringInTime={'werwer'} />
      </BetMatchCell>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it(' toJson  BetButtonCell ', () => {
    const wrapper = render(
      <BetMatchCell
        matchTime={'werwer'}
        prefixCls={'push-inplay'}
        clickBetBtn={() => {
          console.log('test12');
        }}
      >
        <BetMatchInfo
          homeName={'678'}
          homeScore={34}
          awayName={'34'}
          awayScore={34}
          isLive={true}
          isShowScore={true}
          liveText="LIVE"
          time={'werwerwer'}
          clickBetMatchInfo={() => {
            this.goToMatchDetail({ sportType, matchId });
          }}
        />

        <BetButtonCell
          prefixCls={'push-inplay'}
          homeName={'wer'}
          awayName={'werwer'}
          playtypeName={'werwer'}
          playtypeCode={'werwer'}
          matchId={'werwrwer'}
          sportType={'ewrwer'}
          handicapType={true}
          leagueName={'werwer'}
          matchName={'ewrwer'}
          matchType={'werwer'}
          sportName={'wer'}
          period={1}
          passType={'werewr'}
        >
          <BetButton betHandicap={'-1'} isSelected={true} optionName={'32423'} isDetail={false} />
          <Line />
        </BetButtonCell>
      </BetMatchCell>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it(' toJson ProgressText click', () => {
    const wrapper = mount(
      <BetMatchCell
        matchTime={'werwer'}
        prefixCls={'push-inplay'}
        clickBetBtn={() => {
          console.log('test12');
        }}
      >
        <BetMatchInfo
          homeName={'678'}
          homeScore={34}
          awayName={'34'}
          awayScore={34}
          isLive={true}
          isShowScore={true}
          liveText="LIVE"
          time={'werwerwer'}
          clickBetMatchInfo={() => {
            this.goToMatchDetail({ sportType, matchId });
          }}
        />

        <ProgressText text={'ewrwe'} stringInTime={'werwer'} />
      </BetMatchCell>
    );
    wrapper.find('ProgressText').simulate('click');
  });

  it(' toJson  BetButtonCell click ', () => {
    const wrapper = mount(
      <BetMatchCell>
        <BetMatchInfo />

        <BetButtonCell>
          <BetButton />
          <Line />
        </BetButtonCell>
      </BetMatchCell>
    );
    wrapper
      .find('BetButtonCell')
      .find('BetButton')
      .simulate('click');
  });
});
