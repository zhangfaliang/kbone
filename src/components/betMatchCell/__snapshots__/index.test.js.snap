// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`betMatchCell  toJson  BetButtonCell  1`] = `
<div
  class="push-inplay-bet-cell"
>
  <div
    class="wrap"
  >
    <div
      class="nameOrScore"
    >
      <span
        class="name"
      >
        678
      </span>
      <span
        class="score"
      >
        34 
      </span>
    </div>
    <div
      class="nameOrScore"
    >
      <span
        class="name"
      >
        34
      </span>
      <span
        class="score"
      >
        34
      </span>
    </div>
    <div
      class="bottom"
    >
      <span
        class="live"
      >
        LIVE
      </span>
      <span
        class="time"
      >
        werwerwer
      </span>
    </div>
  </div>
  <div
    class="push-inplay-betButtonCell"
  >
    <div
      class="default-bet-button"
    >
      <span
        class="up"
      >
        <span
          class="iconfont icon-sheng"
        />
      </span>
      <span
        class="down"
      >
        <span
          class="iconfont icon-jiang"
        />
      </span>
      <div
        class="wrapSp"
      >
        <span
          class="sp"
        />
      </div>
    </div>
    <div
      class="row-line"
    />
  </div>
</div>
`;

exports[`betMatchCell  toJson ProgressText 1`] = `
<div
  class="push-inplay-bet-cell"
>
  <div
    class="wrap"
  >
    <div
      class="nameOrScore"
    >
      <span
        class="name"
      >
        678
      </span>
      <span
        class="score"
      >
        34 
      </span>
    </div>
    <div
      class="nameOrScore"
    >
      <span
        class="name"
      >
        34
      </span>
      <span
        class="score"
      >
        34
      </span>
    </div>
    <div
      class="bottom"
    >
      <span
        class="live"
      >
        LIVE
      </span>
      <span
        class="time"
      >
        werwerwer
      </span>
    </div>
  </div>
  <div
    class="progress"
  >
    <span
      class="text"
    >
      ewrwe
    </span>
    <span
      class="stringInTime"
    >
      werwer
    </span>
  </div>
</div>
`;
